src/c/inteldeflater/IntelDeflater.c

```diff
@@ -116,20 +201,65 @@ Java_htsjdk_samtools_util_zip_IntelDeflater_setDictionary(JNIEnv *env, jclass cl
 }
 
 JNIEXPORT jint JNICALL
-Java_htsjdk_samtools_util_zip_IntelDeflater_deflateBytes(JNIEnv *env, jobject this, jlong addr,
+Java_net_sf_samtools_util_zip_IntelDeflater_deflateBytes(JNIEnv *env, jobject this, jlong addr,
                                          jarray b, jint off, jint len, jint flush)
 {
-    z_stream *strm = jlong_to_ptr(addr);
+  jarray this_buf = (*env)->GetObjectField(env, this, bufID);
+  jint this_off = (*env)->GetIntField(env, this, offID);
+  jint this_len = (*env)->GetIntField(env, this, lenID);
+  jbyte *in_buf;
+  jbyte *out_buf;
+  LZ_Stream2 *strm;
+  if (useIGZIP)
+    strm = jlong_to_ptr(addr);
+  //igzip only supports one compression level so setParamsID should not be set when using igzip 
+  //igzip does not support flush
+  if (useIGZIP && (((*env)->GetBooleanField(env, this, setParamsID) && strm->total_in != 0)  || flush == 1)) {
+    JNU_ThrowInternalError(env, "igzip doesn't support this");
+  } else if (useIGZIP) {
+    in_buf = (*env)->GetPrimitiveArrayCritical(env, this_buf, 0);
+    if (in_buf == NULL) {
+      // Throw OOME only when length is not zero
+      if (this_len != 0)
```

Comment:
braces around `if`s make it much clearer, or put it on the same line throughout please